examples/service-fastapi/app/security.py

```diff
@@ -0,0 +1,86 @@
+import os
+
+from fastapi import HTTPException
+from fastapi import Security
+from fastapi import status
+from fastapi.security import OAuth2AuthorizationCodeBearer
+from fastapi.security.api_key import APIKeyCookie
+from fastapi.security.api_key import APIKeyQuery
+
+from .client import get_client
+
+### For authenticated endpoints, we want to get auth from one of three ways
+### 1. "token" in the url params
+### 2. "jupyterhub-services" cookie (or config via env)
+### 3. Authorization Bearer header (with oauth to Hub support)
+auth_by_param = APIKeyQuery(name="token", auto_error=False)
+
+COOKIE_NAME = os.getenv("JUPYTERHUB_COOKIE_NAME", "jupyterhub-services")
```

Comment:
I think you can skip the shared-cookie auth. This is on its way out in favor of oauth all the time. It will work, but it's never required.

Reply:
auth by cookie removed